src/psyclone/domain/lfric/lfric_kern.py

```diff
@@ -105,6 +105,7 @@ def __init__(self):
         self._qr_rules = OrderedDict()
         self._cma_operation = None
         self._is_intergrid = False  # Whether this is an inter-grid kernel
+        self._intergrid_ref = None  # Reference to this kernel inter-grid
```

Comment:
Done, but to achieve this I had to move its initialisation from DynMesh to this class setup(), which I think is a good change since we can use 'self' instead external private methods for this functionality.

Reply:
Great :-)